scripts/add_token_list.js

```diff
@@ -1,33 +0,0 @@
-import { addTokens, getBatchExchange, getOwl } from "../build/common/scripts/util"
-const { token_list_url } = require("./utilities.js")
-const fetch = require("node-fetch")
-const argv = require("yargs")
-  .option("token_list_url", {
-    describe: "A url which can be fetched with node-fetch",
-    default: token_list_url,
-  })
-  .help(false)
-  .version(false).argv
-
-module.exports = async function (callback) {
-  try {
-    const tokenList = await (await fetch(argv.token_list_url)).json()
-    const networkId = String(await web3.eth.net.getId())
-
-    const tokenAddresses = []
-    for (const token in tokenList) {
```

Comment:
@nlordell then this must have been a major typo before.

Reply:
Its because its using
```js
tokenList[token]
```

Which is fine because `token` is, as you mentioned, an ID.

Reply:
Yea, sure but still subtle and ugly but why would in just arbitrarily choose one of the entries?

Reply:
In fact, if this list were not sorted by id, or the IDs exceeded the index of the array this would have failed.

Reply:
Also, notice that this implementation does not consider uniqueness of the token addresses.

Reply:
This is becuase `for (const foo in bar)` iterates for all keys in `bar`. This is used for iterating over keys of an object more than array values. For example:
```js
const bar = { name: "answer", value: 42 };
for (const foo in bar) { console.log(foo) }
// # Prints:
// name
// value
```

This also happens to work for arrays as well:
```js
const bar = ["a", "b", "c"];
for (const foo in bar) { console.log(foo) }
// # Prints:
// 0
// 1
// 2
```

Additionally, it works for sparse arrays:
```js
const bar = []7
bar[1337] = 42;
bar[3] = 3
for (const foo in bar) { console.log(foo) }
// # Prints:
// 3
// 1337
```

Reply:
At least its consistent! Thanks!